src/transformers/models/tvp/modeling_tvp.py

```diff
@@ -0,0 +1,960 @@
+
+    def forward(self, x):
+        base = torch.zeros(1, 1, 3, self.base_size, self.base_size)
+        prompt = torch.cat([self.pad_left, base, self.pad_right], dim=4)
+        prompt = torch.cat([self.pad_up, prompt, self.pad_down], dim=3)
+        prompt = torch.cat(x.size(1) * [prompt], dim=1)
+        prompt = torch.cat(x.size(0) * [prompt])
+        return prompt
+
+
+# Pad single images only at the bottom.
+class TvpDownPadPrompter(nn.Module):
+    def __init__(self, image_size, pad_size):
+        super(TvpDownPadPrompter, self).__init__()
+        self.pad_size = pad_size
+        self.image_size = image_size
+
+        self.pad_down = nn.Parameter(torch.randn([1, 1, 3, pad_size, image_size]))
+
+    def forward(self, x):
+        prompt = torch.zeros([x.shape[0], x.shape[1], 3, self.image_size, self.image_size])
+        start_point = self.image_size - self.pad_size
+        prompt[:, :, :, start_point : self.image_size, :] = self.pad_down
+        return prompt
+
+
+# Pad frames extracted from videos only at the bottom.
+class TvpFrameDownPadPrompter(nn.Module):
+    def __init__(self, image_size, pad_size, frame_num):
+        super(TvpFrameDownPadPrompter, self).__init__()
+        self.pad_size = pad_size
+        self.image_size = image_size
+        self.frame_num = frame_num
+
+        self.pad_down = nn.Parameter(torch.randn([1, frame_num, 3, pad_size, image_size]))
+
+    def forward(self, x):
+        prompt = torch.zeros([x.shape[0], x.shape[1], 3, self.image_size, self.image_size])
+        start_point = self.image_size - self.pad_size
+        prompt[:, :, :, start_point : self.image_size, :] = self.pad_down
+        return prompt
+
+
+# Pad frames extracted from videos in the surroundings.
+class TvpFramePadPrompter(nn.Module):
+    def __init__(self, image_size, prompt_size, frame_num):
+        super(TvpFramePadPrompter, self).__init__()
+        pad_size = prompt_size
+        image_size = image_size
+        self.frame_num = frame_num
+
+        self.base_size = image_size - pad_size * 2
+        self.pad_up = nn.Parameter(torch.randn([1, frame_num, 3, pad_size, image_size]))
+        self.pad_down = nn.Parameter(torch.randn([1, frame_num, 3, pad_size, image_size]))
+        self.pad_left = nn.Parameter(torch.randn([1, frame_num, 3, image_size - pad_size * 2, pad_size]))
+        self.pad_right = nn.Parameter(torch.randn([1, frame_num, 3, image_size - pad_size * 2, pad_size]))
+
+    def forward(self, x):
+        base = torch.zeros(1, self.frame_num, 3, self.base_size, self.base_size)
+        prompt = torch.cat([self.pad_left, base, self.pad_right], dim=4)
+        prompt = torch.cat([self.pad_up, prompt, self.pad_down], dim=3)
+        prompt = torch.cat(x.size(0) * [prompt])
+        return prompt
+
+
+prompter_map = {
+    "downpad": TvpDownPadPrompter,
+    "pad": TvpPadPrompter,
+    "framedownpad": TvpFrameDownPadPrompter,
+    "framepad": TvpFramePadPrompter,
+}
```

Comment:
the reason why we pad here and not in the image processor is because the weights are required I guess?

Reply:
Yes, exactly.